### PR TITLE
Access logging fix and enable remote ip

### PR DESCRIPTION
**Pull request by liborfuka**

### CODE REVIEW

File: src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java [hunk 1]
```diff
@@ -48,6 +48,10 @@ public EmbeddedServletContainerFactory servletContainer() {
 
         // Default configuration file: logback-access.xml
         LogbackValve logbackValve = new LogbackValve();
+        
+        logbackValve.setFilename("logback-access.xml");
```
Comment (metacosm) [inline src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java:52]: This shouldn't be needed as logback-access.xml is already the default file name (see comment line 49)

File: src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java [hunk 2]
```diff
@@ -48,6 +48,10 @@ public EmbeddedServletContainerFactory servletContainer() {
...
+
+        tomcat.addContextValves(logbackValve);
```
Comment (metacosm) [inline src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java]: This is done right beneath, line 57.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -3,5 +3,5 @@ server:
     enabled: true
     min-response-size: 1
   use-forward-headers: true
-logging:
-  level.org.apache.tomcat: "DEBUG"
\ No newline at end of file
+  tomcat.protocol-header: x-forwarded-proto
```
Comment (metacosm) [inline src/main/resources/application.yml]: Based on the Spring Boot documentation, this is unneeded. See: https://docs.spring.io/spring-boot/docs/current/reference/html/howto-embedded-servlet-containers.html#howto-use-tomcat-behind-a-proxy-server, the only thing that's needed is to set `server.use-forward-headers` to true. At least, that is my understanding.

**Review by metacosm — CHANGES_REQUESTED** [review]

**Comment by cmoulliard** [comment]
The name of the log file can be defined within the application.properties file and doesn't need to be hard coded --> https://github.com/snowdrop/spring-boot-http-2/blob/master/src/main/resources/application.properties#L12-L19

File: src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java (hunk 1 above)
Comment (liborfuka) [reply src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java:52]: I tested without this line and access logging to output does not work. Try it.

File: src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java (hunk 2 above)
Comment (liborfuka) [reply src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java]: Typo, thanks

File: src/main/resources/application.yml (hunk 3 above)
Comment (liborfuka) [reply src/main/resources/application.yml]: Correct, these two properties are not needed:
tomcat.protocol-header: x-forwarded-proto
tomcat.remote-ip-header: x-forwarded-for
Tested on OCP 3.6.

File: src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java (hunk 1 above)
Comment (metacosm) [reply src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java:52]: OK.

File: src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java (hunk 2 above)
Comment (metacosm) [reply src/main/java/me/snowdrop/logging/TomcatLoggingApplication.java]: Can you update the commits, I'll merge.

**Review by metacosm — APPROVED** [review]

**Pull request merged**